base/constants_syncdocs.go

```diff
@@ -457,10 +468,51 @@ func SetSyncInfoMetaVersion(ds DataStore, metaVersion string) error {
 	return err
 }
 
-// SerializeIfLonger returns name as a sha1 string if the length of the name is greater or equal to the length specificed. Otherwise, returns the original string.
+// SerializeIfLonger returns name as a sha1 string if the length of the name is greater or equal to the length specified. Otherwise, returns the original string.
 func SerializeIfLonger(name string, length int) string {
 	if len(name) < length {
 		return name
 	}
 	return Sha1HashString(name, "")
 }
+
+// CompareMetadataVersion Will build comparable build version for comparison with meta version defined in syncInfo, then
+// will return true if we require attachment migration, false if not.
+func CompareMetadataVersion(ctx context.Context, metaVersion string) (bool, error) {
+	var syncInfoVersion *ComparableBuildVersion
+	var requiresAttachmentMigration bool
+	var err error
+	if metaVersion != "" {
+		syncInfoVersion, err = NewComparableBuildVersionFromString(metaVersion)
+		if err != nil {
+			return true, err
+		}
+		requiresAttachmentMigration, err = CheckRequireAttachmentMigration(ctx, syncInfoVersion)
+		if err != nil {
+			return true, err
+		}
+	} else {
+		// no meta version passed in, thus attachment migration should take place
+		requiresAttachmentMigration = true
+	}
+	return requiresAttachmentMigration, nil
+}
```

Comment:
nit: this is probably easier to read like this:

```suggestion
    // no meta version passed in, thus attachment migration should take place
    if metaVersion == "" {
    	return requiresAttachmentMigration, nil
    }
	syncInfoVersion, err := NewComparableBuildVersionFromString(metaVersion)
	if err != nil {
		return true, err
	}
	return CheckRequireAttachmentMigration(ctx, syncInfoVersion)
}
```